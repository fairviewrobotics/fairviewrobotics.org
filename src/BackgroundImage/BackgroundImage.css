#image-parallax {
  height: calc(100% + 100px);
  width: 100%;
  top: -100px;
  position: fixed;
  z-index: -1;
  background: black 50% 0;
  background-size: cover;
  transition: background-image 1s, background-size 0s;
  transition-delay: 2s;
}

#image-parallax.not-fullscreen {
  height: calc(25vw + 100px) !important;
}

.intro-mask {
  position: absolute;
  width: 100%;
  left: 0;
  top: 0;
  height: 100%;
  margin: 0;
  background-color: rgba(255, 255, 255, 0.6);
}

.intro-mask.not-fullscreen {
  height: calc(25vw) !important;
}

@media(max-width:768px) {
  #image-parallax.not-fullscreen {
    height: calc(35vw + 100px) !important;
  }

  .intro-mask.not-fullscreen {
    height: calc(35vw) !important;
  }
}