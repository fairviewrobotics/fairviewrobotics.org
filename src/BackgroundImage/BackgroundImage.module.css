.imageParallax {
  height: calc(100% + 100px);
  width: 100%;
  top: -100px;
  position: fixed;
  z-index: -1;
  background: black 50% 0;
  background-size: cover;
  transition: background-image 1s, background-size 0s;
  transition-delay: 2s;
}

.imageParallax.notFullscreen {
  height: calc(25vw + 100px) !important;
}

.introMask {
  position: absolute;
  width: 100%;
  left: 0;
  top: 0;
  height: 100%;
  margin: 0;
  background-color: rgba(255, 255, 255, 0.6);
}

.introMask.notFullscreen {
  height: calc(25vw) !important;
}

@media(max-width:768px) {
  .imageParallax.notFullscreen {
    height: calc(35vw + 100px) !important;
  }

  .introMask.notFullscreen {
    height: calc(35vw) !important;
  }
}