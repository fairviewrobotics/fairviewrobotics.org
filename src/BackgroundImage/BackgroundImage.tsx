import React, { Fragment } from 'react';
import styles from './BackgroundImage.module.css';

const BackgroundImage = ({ src, fullScreen, animate }: { src: string, fullScreen: boolean, animate?: { duration: number, delay: number } }) => {
  const imageStyle = {
    backgroundImage: `url(${src})`
  };

  const animationStyle = animate ?
    {
      transition: `background-image ${animate.duration}ms, background-size 0s`,
      transitionDelay: `${animate.delay}ms`
    }
    :
    null;

  const parallaxStyle = {...animationStyle, ...imageStyle};

  let backgroundImage = (
    <Fragment>
      <div className={styles.introMask} />
      <div className={styles.imageParallax} style={parallaxStyle} />
    </Fragment>
  );

  if (!fullScreen) {
    backgroundImage = (
      <div className={styles.imageContainer}>
        {backgroundImage}
      </div>
    );
  }

  return backgroundImage;
};

BackgroundImage.defaultProps = {
  fullScreen: false
};

export default BackgroundImage;