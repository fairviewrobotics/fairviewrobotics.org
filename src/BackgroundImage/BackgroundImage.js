import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import './BackgroundImage.css';

export default class BackgroundImage extends PureComponent {

  static propTypes = {
    src: PropTypes.string.isRequired,
    fullScreen: PropTypes.bool,
    animate: PropTypes.shape({
      duration: PropTypes.number.isRequired,
      delay: PropTypes.number.isRequired
    })
  };

  static defaultProps = {
    fullScreen: false
  };

  render() {
    const imageStyle = {
      backgroundImage: `url(${this.props.src})`
    };

    if (!this.props.fullScreen) {
      imageStyle.position = 'absolute';
      imageStyle.backgroundPosition = '50% 75px';
    }
    const fullScreenClass = !this.props.fullScreen ? 'not-fullscreen' : '';
    
    const animationStyle = this.props.animate ?
      {
        transition: `background-image ${this.props.animate.duration}ms, background-size 0s`,
        transitionDelay: `${this.props.animate.delay}ms`
      }
      :
      null;

    return (
      <div>
        <div className={`intro-mask ${fullScreenClass}`} />
        <div id="image-parallax" className={fullScreenClass} style={{ ...imageStyle, ...animationStyle }} />
      </div>
    );
  }
}
