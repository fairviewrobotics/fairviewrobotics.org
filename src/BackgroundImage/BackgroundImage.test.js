import React from "react";
import { shallow } from "enzyme";
import BackgroundImage from "./BackgroundImage";

const setup = (props = {}) => {
  const defaultProps = {
    src: 'http://urltoimage.com/image.jpg'
  };
  const finalProps = {...defaultProps, ...props};

  const wrapper = shallow(<BackgroundImage {...finalProps} />);

  return {
    wrapper,
    imageParallax: wrapper.find('.imageParallax')
  };
};

describe('BackgroundImage', () => {
  it('renders without crashing', () => {
    setup();
  });

  describe('when given a `src`', () => {
    it('renders an element with a background of `src`', () => {
      const src = 'whackyMeme.jpg';

      const { wrapper } = setup({ src });

      expect(wrapper).toMatchSnapshot();
      // expect(wrapper.find('.imageParallax')).toHaveStyle('backgroundImage', `url(${src})`)
    });
  });

  describe('when not `fullScreen`', () => {
    it('renders a image container for the background image', () => {
      const { wrapper } = setup();

      expect(wrapper.find('.imageContainer')).toExist();
    });
  });

  describe('when `animate`', () => {
    let image;

    beforeEach(() => {
      const { imageParallax } = setup({
        animate: {
          duration: 1000,
          delay: 500
        }
      });

      image = imageParallax;
    });

    it('renders the background image with a transition with a `duration`', () => {
      expect(image.prop('style').transition).toMatchSnapshot();
    });

    it('renders the background image with a transition `delay`', () => {
      expect(image.prop('style').transitionDelay).toMatchSnapshot();
    });
  });

});