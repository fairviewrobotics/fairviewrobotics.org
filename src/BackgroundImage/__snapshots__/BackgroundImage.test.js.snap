// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackgroundImage when \`animate\` renders the background image with a transition \`delay\` 1`] = `"500ms"`;

exports[`BackgroundImage when \`animate\` renders the background image with a transition with a \`duration\` 1`] = `"background-image 1000ms, background-size 0s"`;

exports[`BackgroundImage when given a \`src\` renders an element with a background of \`src\` 1`] = `
<div
  className="imageContainer"
>
  <React.Fragment>
    <div
      className="introMask"
    />
    <div
      className="imageParallax"
      style={
        Object {
          "backgroundImage": "url(whackyMeme.jpg)",
        }
      }
    />
  </React.Fragment>
</div>
`;
